cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SQLConn.scala

```diff
@@ -0,0 +1,97 @@
+package edu.uci.ics.cloudberry.zion.model.impl
+import edu.uci.ics.cloudberry.zion.model.datastore.IDataConn
+
+import play.api.libs.ws.WSResponse
+import play.api.libs.json.Json
+import play.api.libs.json._
+
+import scala.concurrent.{ExecutionContext, Future}
+
+import java.sql.{Connection, DriverManager}
+import java.io.InputStream
+import java.util.Date
+import java.text.SimpleDateFormat
+
+class SQLConn(url: String)(implicit ec: ExecutionContext) extends IDataConn {
+  val driver = "com.mysql.jdbc.Driver"
+  val username = "root"
+  val password = "2048"
+  var connection:Connection = _
+  Class.forName(driver)
+  connection = DriverManager.getConnection(url, username, password)
+  val defaultQueryResponse = Json.toJson(Seq(Seq.empty[JsValue]))
+  val stream : InputStream = getClass.getResourceAsStream("/ddl/berry.json")
+  val source = scala.io.Source.fromInputStream(stream).getLines.mkString
+  val statement = connection.createStatement
+  val berry: JsValue = Json.parse(source)
+
+  def post(query: String): Future[WSResponse] = {
+    throw new UnsupportedOperationException
+  }
+
+  def postQuery(query: String): Future[JsValue] = {
+    val result = statement.executeQuery(query)
+    val rsmd = result.getMetaData
+    val columnCount = rsmd.getColumnCount
+    var qJsonArray: JsArray = Json.arr()
+    if (!result.isBeforeFirst()) {
+      Future(true)
+    }
+    try {
+      while (result.next) {
+        var index = 1
+        var rsJson: JsObject = Json.obj()
+        while (index <= columnCount) {
+          val columnLabel = rsmd.getColumnLabel(index)
+          val value = result.getObject(columnLabel)
```

Comment:
This Try Catch is not neat.. But I don't know how to break a loop in scala when value = result.getObject(columnLabel) returns null.
Any suggestions?